tests/integ/scala/test_literal_suite.py

```diff
@@ -122,11 +135,18 @@ def test_special_literals(session):
         .with_column("literal", lit(source_literal))
     )
 
-    assert (
-        str(df.schema) == "StructType([StructField('ID', LongType(), nullable=False), "
-        "StructField('NULL', StringType(), nullable=True), "
-        "StructField('LITERAL', LongType(), nullable=False)])"
-    )
+    if session.sql_simplifier_enabled:
+        assert (
+            str(df.schema) == "StructType([StructField('ID', LongType(), nullable=False), "
+            "StructField('NULL', StringType(), nullable=True), "
+            "StructField('LITERAL', LongType(), nullable=False)])"
+        )
+    else:
+        assert (
+            str(df.schema) == "StructType([StructField('ID', LongType(), nullable=False), "
+            "StructField('NULL', StringType(16777216), nullable=True), "
```

Comment:
@sfc-gh-yixie can this be considered behavior change? In case when sql simplified is disabled, in this test we compare `str(schema)` instead of `schemas`. With the updated `__eq__` comparing schemas still evaluates to be equal